islandora_plupload.module

```diff
@@ -55,6 +55,33 @@ function islandora_plupload_alter_managed_files(&$form) {
       $element =& $form[$element_key];
       islandora_plupload_alter_managed_files($element);
       if (isset($element['#type']) && $element['#type'] === 'managed_file' && !(isset($element['#islandora_plupload_do_not_alter']) && $element['#islandora_plupload_do_not_alter'])) {
+        // The Plupload and Islandora Plupload modules create a bug that
+        // prevents users from adding datastreams and replacing datastreams that
+        // allow any file type as a result of validating a files extension
+        // (ex. .jpg) against an empty string.
+        $current_path = current_path();
+        if (strpos($current_path, 'datastreams/add')) {
+          $element['#process'] = array(
+            'islandora_plupload_element_processs',
+          );
+        }
+        elseif (preg_match('/\/datastream\/.*\/replace/', $current_path)) {
```

Comment:
Same deal here, if using `menu_get_item()` could match directly against `$menu_item['path'] == 'islandora/object/%/datastream/%/replace'`. It'd also allow you to directly grab the datastream (and therefore DSID/parent) from looking at `$menu_item['page_arguments'][1]`